src/main/java/gomedic/model/person/AbstractPerson.java

```diff
@@ -0,0 +1,83 @@
+package gomedic.model.person;
+
+import java.util.Objects;
+
+import gomedic.commons.util.CollectionUtil;
+import gomedic.model.commonfield.Name;
+import gomedic.model.commonfield.Phone;
+
+/**
+ * Represents a AbstractPerson in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public abstract class AbstractPerson {
```

Comment:
by the way, I think it would be nice if we just call it `Person` instead of `AbstractPerson`, the `Abstract`ness can be represented in UML diagram later instead of the class name

Reply:
Yeap sure, for now I'm just using `AbstractPerson` as a temporary placeholder name cos `Person` is used by the original AB3, so I am just keeping the original one for now to pass the AB3 tests